ush/HWP_tools.py

```diff
@@ -25,6 +38,22 @@ def check_restart_files(hourly_hwpdir, fcst_dates):
     return(hwp_avail_hours, hwp_non_avail_hours)
 
 def copy_missing_restart(nwges_dir, hwp_non_avail_hours, hourly_hwpdir):
+    """Check the restart files.
+
+    Args:
+         nwges_dir: holds the boundary, INPUT, and RESTART files (e.g,
+         described here:
+         [ufs-community/ufs-srweather-app#654](https://github.com/ufs-community/ufs-srweather-app/issues/654))
+         The use of NWGES is a bit of a legacy variable name in
+         production. It is often used to define a directory holding
+         DA-relevant files (such as first guess and analysis files)
+         outside of the main COM output directory.
+         hwp_non_avail_hours: a list of hours without available RESTART files
+         hourly_hwpdir: the directory where the RESTART data is copied (nwges/HOURLY_HWP)
+    Returns:
+         list of (restart_avail_hours, restart_nonavail_hours_test)

```

Comment:
@jordanschnell last question: what do these mean:
(restart_avail_hours, restart_nonavail_hours_test)

Reply:
restart_avail_hours and restart_nonavail_hours_test is a subset of hwp_non_avail_hours. The script loops through hwp_non_avail_hours (missing files in nwges/HOURLY_HWP) and determines if an alternative RESTART file in nwges/fcst_fv3lam/RESTART is available instead. If so, it is appended to restart_avail_hours.

Reply:
Thanks, that explains things very well.

@jordanschnell I have add all these answers to the file. Please take a moment to review to make sure I got everything correct. If you want to add more detail, that would be very welcome.